docker/auth/auth.py

```diff
@@ -182,13 +191,14 @@ def parse_auth(entries, raise_on_error=False):
                 'Auth data for {0} is absent. Client might be using a '
                 'credentials store instead.'
             )
-            return {}
+            conf[registry] = {}
```

Comment:
Technically, either the client is using a cred store and all entries are empty, or they're not and none should be, so returning an empty dict here is not a big problem - but just in case weird hybrid `config.json` files start popping up, this should allow it to continue working.